pkg/storaged/format-dialog.jsx

```diff
@@ -106,25 +106,30 @@ export function mounting_dialog_options(vals) {
     return unparse_options(opts);
 }
 
-export function crypto_options_dialog_fields(options, visible) {
+export function crypto_options_dialog_fields(options, visible, include_store_passphrase) {
     var split_options = parse_options(options);
     var opt_auto = !extract_option(split_options, "noauto");
     var opt_ro = extract_option(split_options, "readonly");
     var extra_options = unparse_options(split_options);
 
+    var fields = [
+        { title: _("Unlock at boot"), tag: "auto" },
+        { title: _("Unlock read only"), tag: "ro" },
+        { title: _("Custom encryption options"), tag: "extra", type: "checkboxWithInput" },
+    ];
+
+    if (include_store_passphrase)
+        fields = [ { title: _("Store passphrase"), tag: "store_passphrase" } ].concat(fields);
```

Comment:
OOI, what does that actually do? This feels a bit dangerous, like "my passphrase will be written in plaintext somewhere". This should perhaps at least get a tooltip with some explanation? (separate PR is fine, it's nothing new)

Reply:
> OOI, what does that actually do? This feels a bit dangerous, like "my passphrase will be written in plaintext somewhere".

This is exactly what happens.  The passphrase gets stored into a file in /etc/luks-keys/ and used automatically during boot.  This makes sense for external storage attached via USB or some NAS thing, I guess.  (This is a feature of cryptsetup and UDisks2, so I made a UI for it.... but it mostly creates confusion...)

Reply:
Ack, thanks. Unrelated to this PR, so resolving. I filed issue #11824 about it.